examples/Q5_isingmodel-NumPy.py

```diff
@@ -0,0 +1,88 @@
+# Qubit Optimization for 3 qubit Ising model using  Pennylane default qubit ecosystem.
+
+# Ising models are used in Quantum Annealing to solve Quadratic Unconstrained Binary
+
+# Optimization problems with non-convex  cost functions. This example demonstrates how
+
+# Gradient descent optimizers can get stuck in local minima when using non-convex cost functions.
+
+# This example  uses and compares pennylane gradient descent optimizer with the optimizers
+
+# of PyTorch and and TensorFlow for this quantum system: see Q5_isingmodel.ipynb
+
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.optimize import GradientDescentOptimizer
+
+# making a device with pennylane dafault qubit command
+
+dev1 = qml.device("default.qubit", wires=3)
+
+# we will be solving the problem with fixed coupling between the three
+# qubits and optimizing the phase operator to get minimum energy
+# configuration of the spin system
+
+
+@qml.qnode(dev1)
+def circuit1(p1, p2):
+    # 	lets say first spin is always up (+1 eigenstate of pauli-z operator)
+    # 	then we can optimize the rotation angles for the other two spins
+    # 	so that the energy in minimized for the given couplings
+
+    # 	WE USE THE GENERAL Rot(phi,theta,omega,wires) single qubit operation
+
+    qml.Rot(p1[0], p1[1], p1[2], wires=1)
+    qml.Rot(p2[0], p2[1], p2[2], wires=2)
+    return qml.expval.PauliZ(0), qml.expval.PauliZ(1), qml.expval.PauliZ(2)
+
+
+def cost(var):
+    # 	let coupling matrix be J=[1, -1]
+    J = np.array([1, -1])
+    #       circuit1 function returns a numpy array of pauliz exp values
+    spins = circuit1(var[0], var[1])
+    # 	the expectation value of pauliZ is plus 1 for spin up and -1 for spin down
+    energy = -sum(J_ij * spins[i] * spins[i + 1] for i, J_ij in enumerate(J))
+    return energy
+
+
+# TEST RUN: test it for [1,-1,-1] spin configuration. Total energy for this ising model should be
+
+# H = -1(J1*s1*s2 + J2*s2*s3) = -1 ( 1*1*-1  +  -1*-1*-1 ) = 2
+
+test1 = np.array([0, np.pi, 0])
+test2 = np.array([0, np.pi, 0])
+cost_check = cost([test1, test2])
+print(cost_check)
```

Comment:
It might be nice to include more information in these print statement for the user in the terminal, something like
```python
print("Energy for [1, -1, -1] spin configuration:", cost_check)
```
(This can also be done for all following print statements)